src/ucp/core/ucp_worker.c

```diff
@@ -610,6 +655,7 @@ void ucp_worker_destroy(ucp_worker_h worker)
     ucs_trace_func("worker=%p", worker);
     ucp_worker_remove_am_handlers(worker);
     ucp_worker_destroy_eps(worker);
+    ucs_mpool_cleanup(&worker->am_mp, 1);
```

Comment:
Really yes.. but this possible only for 2 cases:

1. num_tls == 0
2. max possible msg == 0 

Both cases look erroneous and should be handled earlier, right?
I'll fix it by creation mpool always at least with rx_headroom size